plugins/content-manager/src/main/java/com/wazuh/contentmanager/client/CTIClient.java

```diff
@@ -184,6 +230,101 @@ public static Map<String, String> contextQueryParameters(
         return params;
     }
 
+    /**
+     * Send a request to the CTI API and handles the HTTP response based on the provided status code.
+     *
+     * @param function The function to be executed.
+     * @param method The HTTP method to use for the request.
+     * @param endpoint The endpoint to append to the base API URI.
+     * @param body The request body (optional, applicable for POST/PUT).
+     * @param params The query parameters (optional).
+     * @param header The headers to include in the request (optional).
+     * @param maxAttempts The number of remaining attempts.
+     * @throws IOException If an error occurs during response processing.
+     */
+    private SimpleHttpResponse fetchWithRetry(
+            Functions function,
+            Method method,
+            String endpoint,
+            String body,
+            Map<String, String> params,
+            Header header,
+            int maxAttempts) {
+
+        int attemptsLeft = maxAttempts;
+        ZonedDateTime cooldown = null;
+
```

Comment:
The value of `maxAttempts` can be replaced by `CTIClient.MAX_ATTEMPTS`
```suggestion
            ) {

        int attemptsLeft = CTIClient.MAX_ATTEMPTS;
        ZonedDateTime cooldown = null;

```

Reply:
Resolve on [eab411d](https://github.com/wazuh/wazuh-indexer-plugins/pull/363/commits/eab411d86a730ead1c56e4b8c211b2248997e663)